website/programs.py

```diff
@@ -205,46 +207,43 @@ def save_program(self, user):
             "achievements": self.achievements.get_earned_achievements(),
         })
 
-    @route("/share", methods=["POST"])
+    @route('/share', methods=['POST'], defaults={'loop_index': 0, 'second_teachers_programs': False})
```

Comment:
Even if this endpoint is reached, it'll complain about several thinsg; missing arguement and then the fact that you're retrieving an id of none. To mitigate that,  see my last comment in app.ts!